src/prefect/engine/result/gcs_result.py

```diff
@@ -0,0 +1,137 @@
+import base64
+from typing import TYPE_CHECKING, Any, Optional
+
+import cloudpickle
+
+from prefect.engine.result.base import Result
+from prefect.client import Secret
+from prefect.utilities import logging
+
+if TYPE_CHECKING:
+    import google.cloud
+
+
+class GCSResult(Result):
+    """
+    Result that is written to and read from a Google Cloud Bucket.
+
+    To authenticate with Google Cloud, you need to ensure that your flow's
+    runtime environment has the proper credentials available
+    (see https://cloud.google.com/docs/authentication/production for all the authentication options).
+
+    You can also optionally provide the name of a Prefect Secret containing your
+    service account key.
+
+    Args:
+        - value (Any, optional): the value of the result
+        - bucket (str): the name of the bucket to write to / read from
+        - credentials_secret (str, optional): the name of the Prefect Secret
```

Comment:
Have you tested this with both authentication methods? Would be nice to have an example Prefect Secret in the class docstring

Reply:
indeed, with the default credentials from an already auth'd box and with a Prefect secret containing a service account JSON key --both worked.  I'm not certain by what you mean for an example secret, as it would only be a name of a secret that is already stored in a Prefect Cloud Secret.

Reply:
By example I mean the format of the Secret, either written out like:
```
{
    'secret_thing1': '...',
    'secret_thing2': '...',
    etc.
}
```
or a link to the format of the credentials secret from the GCP docs